src/ripple/proto/org/xrpl/rpc/v1/transaction.proto

```diff
@@ -27,6 +27,16 @@ message Transaction
 
         AccountDelete account_delete = 14;
 
+        AMMInstanceCreate amminstance_create = 41;
+
+        AMMDeposit ammdeposit = 42;
+
+        AMMWithdraw ammwithdraw = 43;
+
+        AMMVote ammvote = 44;
+
+        AMMBid ammbid = 45;
```

Comment:
This whole part of the code has been deprecated. I would prefer to not add these, but some of the unit tests might fail.

Reply:
@cjcobb23

> This whole part of the code has been deprecated. I would prefer to not add these, but some of the unit tests might fail.

Precisely.  This dead code has been in the server for a while now.  Can you please work with someone to remove all of the gRPC serialization code (not just the unit tests) from the rippled code base correctly and without harming Clio?  It sounds like Clio may be in a stable enough place that this problem can finally be addressed.

Reply:
Personally, I'd like to remove gRPC in its entirety and extract data over https or websocket. It seems much easier to remove in its entirety versus having gRPC as a dependency while whittling down gRPC functionality.

This wouldn't be a huge challenge, removes something like 15k Lines of Code from rippled, and can reuse a ton of http/s server code in rippled. 

This is not a technically hard task, it just touches a lot of code and requires familiarity with Clio. I think writing this branch took me 20ish hours of work https://github.com/natenichols/rippled/tree/remove-grpc, but that's with no tests, no polish, and without rewriting Clio. 

It's also challenging because `reporting mode` extracts data over gRPC. We'd either have to rewrite `ETLSource` in rippled, or deprecate reporting mode before we could entirely remove gRPC.

Reply:
Thanks for your research, @natenichols.

I hear you about the preferred path being to remove gRPC all together.  And I hope we get there eventually.  I'm under the impression that once Clio is fully battle tested then deprecating reporting mode should be possible.  But, personally, I'd prefer not to wait an indefinite time for a perfect solution when we have a problem right now: maintaining the gRPC serializer.

What I would hope for is that, with @cjcobb23's guidance, the serializer potion of the rippled gRPC implementation could be removed very soon.  My understanding has been that reporting mode has no dependency on the serializer.  But I could be wrong on that.  I look to you and @cjcobb23 for the accuracy of that.

Once the serializer code is gone I anticipate that it becomes much more obvious what work needs to be done when new transactions and ledger types are added to the code base.

But someone needs to be tasked with that work and given guidance from @cjcobb23.